### PR TITLE
Toggle JSON unfolding by field (again)

### LINKED ISSUE #24801: Allow hiding the whole JSON field at once
**Is your feature request related to a problem? Please describe.**
There is a json field in my database that I do not want to be queryable by user, it is only used for internal purposes on our db. But if I try to hide the json field, I still have to hide the keys of json separately. Also, the structure of that raw json field keeps changing for us, so that means we have to come back to data model again and hide those keys separately too, which is a lot of work given the structure keeps changing very often.

**Describe the solution you'd like**
Being able to hide the json field itself and all it's nested fields at once, since they all fall under the same json column, there should be a way to hide all descendant fields shown by just one toggle.

**How important is this feature to you?**
This is important for me because the internal json field we have in our database has lots of keys present, and not being able to hide them all causes a lot of clutter in all the data that is retrieved through the questions as well as the filter interface.

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by calherries**
This PR is modified from https://github.com/metabase/metabase/pull/28742, which was merged into master but the commit 
had to be reverted ([revert PR](https://github.com/metabase/metabase/pull/29720)).

### Difference between this and #28742:

[This commit](https://github.com/metabase/metabase/pull/29751/commits/c44a4d17936245c845a14425a0e27c892af1da75) contains the changes. Previously the migration to fill in `metabase_field.json_unfolding` was implemented in SQL, and a UDF was created for H2 to parse JSON from `metabase_database.details`. Now, the migration has been converted into a clojure migration, which is simpler because it is the same for all DBMSs and it doesn't need an H2 UDF.

This is a simpler approach, and it also nicely resolves the issue which caused the PR to be reverted, because it removes the dependency on org.json.simple.

1. I've changed how the Unfold JSON setting on the field interacts with the setting at the database level. Turning the toggle off now disables JSON unfolding for the whole database, even if it's enabled for an individual setting. The reason for this change is users are likely to want to be able to kill JSON unfolding for a database in one go. If the DB-level setting is disabled, then the field-level setting doesn't appear in the field settings.
2. Previously the migration to fill in `metabase_field.json_unfolding` was implemented in SQL, and a UDF was created for H2 to parse JSON from `metabase_database.details`. Now, `metabase_database.details` is not needed for the migration because `json_unfolding = true` regardless since `metabase_database.details` is an override, not a default.

### Original PR description:

The PR resolves https://github.com/metabase/metabase/issues/24801 and milestone 1 of https://github.com/metabase/metabase/issues/28667.

I created a new column in the metabase_field table: json_unfolding. The column is false by default for non-JSON columns, and for existing JSON fields, it should take the value of (get-in database [:details :json-unfolding]) or true if it's not set. This means json_unfolding will be true for a field if the database type of the column is 'type/JSON' and nested-field-columns is not supported by a driver. This is by design, so that JSON fields will be automatically unfolded if nested-field-columns is eventually supported by a driver, without the need for a migration on the json_unfolding column. None of the 3rd party drivers [listed in the Metabase docs](https://www.metabase.com/docs/latest/developers-guide/partner-and-community-drivers) support nested-field-columns anyway though.

To change json_unfolding for a field, I've created a new field setting that is visible if the base_type of a field is type/JSON and the driver supports nested-field-columns:
image.

If "Yes", then the JSON field will be unfolded on the next sync. If "No", then any unfolded JSON fields from that field will be inactivated on the next sync. The next milestone will make the sync automatically apply when the setting changes.

You can test this feature out with the json test dataset at test/metabase/test/data/dataset_definitions/json.edn. To create the DB run a test that uses it like json-unfolding-default-true-test with postgres or mysql.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29751)

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
@calherries when you get all other tests passing, please let me know so that I can try to build from the latest commit in this PR. Thanks

**Comment by calherries** [comment]
@nemanjaglumac This is a new PR for the PR you reverted. I ran the Build + Docker Uberjar workflow [here](https://github.com/metabase/metabase/actions/runs/4595332079), and it passed this time. Is there any other tests I need to do before merging to master?

**Comment by nemanjaglumac** [comment]
@calherries that's amazing. Thank you.
As soon as you get the approval from the backend developers, feel free to merge it. 🚢

File: resources/migrations/000_migrations.yaml
Comment (dpsutton) [inline resources/migrations/000_migrations.yaml:14315]: why remove this?

**Review by dpsutton — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29751)* status: 0 of 39 files reviewed, 7 unresolved discussions (waiting on @calherries)

___
*[`resources/migrations/000_migrations.yaml` line 14365 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7fR7_CMLCOZmdPeNJ:-NS7fR7_CMLCOZmdPeNK:b-gping5) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/resources/migrations/000_migrations.yaml#L14365)):*
> ```YAML
> 
>   - changeSet:
>       id: v47.00-002
> ```

There was already an existing v47.00-002 right? Do we need the `validCheckSum: ANY` here?
___
*[`src/metabase/db/custom_migrations.clj` line 160 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7il1l3IIRgoMlChN8:-NS7il1l3IIRgoMlChN9:b-bv9ind) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/db/custom_migrations.clj#L160)):*
> ```Clojure
>                                               (false? (:json-unfolding (json/parse-string details true))))))
>                                   (map :id))
>         field-ids-to-update  (->> (t2/query {:select [:f.id]
> ```

this could potentially be an enormous number of fields. Can we let the db do this instead of bringing it over in Clojure and then sending a huge number back over? 

for pg it would be something like

```clojure
(let [disabled-db-ids #{1 2}] ;; still get the disabled db ids from the first query.
  (t2/query {:update [:metabase_field :f]
             :from   [:metabase_table :t]
             :set    {:json_unfolding true}
             :where  [:and
                      [:= :t.id :f.table_id]
                      [:= :f.base_type "type/JSON"]
                      (when (seq disabled-db-ids)
                        [:not-in :t.db_id disabled-db-ids])]}))
```
___
*[`src/metabase/driver/mysql.clj` line 54 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7peyiDe6jcWH1Ye5S:-NS7peyiDe6jcWH1Ye5T:b-6ycqhf) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/driver/mysql.clj#L54)):*
> ```Clojure
> ;; And MariaDB doesn't have the JSON type at all, though `JSON` was introduced as an alias for LONGTEXT in 10.2.7.
> ;; But since JSON unfolding will only apply columns with JSON types, this won't cause any problems during sync.
> (defmethod driver/database-supports? [:mysql :nested-field-columns] [_driver _feat _db] true)
> ```

You used to be able to disable json unfolding on the whole database. Now it seems like you cannot. Why the change? If someone wants to turn this off, "turn it off for each json field in your db" seems like not a great user experience.
___
*[`src/metabase/driver/postgres.clj` line 56 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7qdn58fWtrzVMCJ9x:-NS7qdn58fWtrzVMCJ9y:b-b5clle) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/driver/postgres.clj#L56)):*
> ```Clojure
> 
> (defmethod driver/database-supports? [:postgres :nested-field-columns]
>   [_driver _feat _db]
> ```

again the same question. Why did the ability to disable on a per-db basis go away?
___
*[`src/metabase/models/database.clj` line 359 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7rhBrBY8aCVnQR7kU:-NS7rhBrBY8aCVnQR7kV:b-z4cued) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/models/database.clj#L359)):*
> ```Clojure
>   ;; nested-field-columns for drivers in the future and have json-unfolding enabled by default, without
>   ;; needing a migration to add the `json-unfolding=true` key to the database details.
>   ((fnil identity true) (get-in database [:details :json-unfolding])))
> ```

i think this is far easier to understand as `(get-in database [:details :json-unfolding] true)` without the fnil.

Also, the `json-unfolding-default` is a bit cumbersome. We have a similar dilemma in persistence: we want to know if (a) the driver supports it, and (b) if this particular instance has it _enabled_.

```clojure
(defmethod driver/database-supports? [:postgres :persist-models]
  [_driver _feat _db]
  true)

(defmethod driver/database-supports? [:postgres :persist-models-enabled]
  [_driver _feat db]
  (-> db :options :persist-models-enabled))
```

I wonder if a similar pattern might help here? You are doing essentially the same thing but using a `json-unfolding-default` function instead of the `database-supports` feature. 
___
*[`test/metabase/api/field_test.clj` line 756 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7uxZhBlUXsq8PA6NN:-NS7uxZhBlUXsq8PA6NO:beswmr4) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/test/metabase/api/field_test.clj#L756)):*
> ```Clojure
>               (sync/sync-database! database)
>               (let [field (t2/select-one Field :id (mt/id :json :json_bit))
>                     enable-json-unfolding! (fn [v]
> ```

this is called `enable-json-unfolding` but the argument dictates if it is enabled or not. note the strange convention of `(enable-json-unfolding! false)`. maybe rename to `set-json-unfolding`

**Review by dpsutton — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29751)* status: 0 of 39 files reviewed, 7 unresolved discussions (waiting on @calherries)

___
*[`src/metabase/driver/mysql.clj` line 54 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7peyiDe6jcWH1Ye5S:-NS83LV6CCrMFTmNjeWv:bg4jrjy) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/driver/mysql.clj#L54)):*
<details><summary><i>Previously, dpsutton wrote…</i></summary><blockquote>

You used to be able to disable json unfolding on the whole database. Now it seems like you cannot. Why the change? If someone wants to turn this off, "turn it off for each json field in your db" seems like not a great user experience.
</blockquote></details>

i see. i'm not sure how i feel about the `json-unfolding-default`. It's far away from here, has a completely separate name that is not discoverable from this one, etc. Props that it's only implemented once though. Tricky

**Review by calherries — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29751)* status: 0 of 39 files reviewed, 7 unresolved discussions (waiting on @dpsutton)

___
*[`resources/migrations/000_migrations.yaml` line 0 at r8](https://reviewable.io/reviews/metabase/metabase/29751#gh-1156395651:-NS9urxg0ZOzflGXP5B1:b-wauoie) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/resources/migrations/000_migrations.yaml#L0)):*
<details><summary><i>Previously, dpsutton wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

why remove this? 
</blockquote></details>

Just cleaning up from my last PR. I was using it during development to update the SQL. Now it's unnecessary
___
*[`resources/migrations/000_migrations.yaml` line 14365 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7fR7_CMLCOZmdPeNJ:-NS9vGr-Eb_VOuj0YPU-:bhrmx00) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/resources/migrations/000_migrations.yaml#L14365)):*
<details><summary><i>Previously, dpsutton wrote…</i></summary><blockquote>

There was already an existing v47.00-002 right? Do we need the `validCheckSum: ANY` here?
</blockquote></details>

I'm not sure what you mean by "existing". Existing where? This is a new migration.
___
*[`src/metabase/db/custom_migrations.clj` line 160 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7il1l3IIRgoMlChN8:-NSA9njy3XUd-B3GXoNu:b-i0gkp1) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/db/custom_migrations.clj#L160)):*
<details><summary><i>Previously, dpsutton wrote…</i></summary><blockquote>

this could potentially be an enormous number of fields. Can we let the db do this instead of bringing it over in Clojure and then sending a huge number back over? 

for pg it would be something like

```clojure
(let [disabled-db-ids #{1 2}] ;; still get the disabled db ids from the first query.
  (t2/query {:update [:metabase_field :f]
             :from   [:metabase_table :t]
             :set    {:json_unfolding true}
             :where  [:and
                      [:= :t.id :f.table_id]
                      [:= :f.base_type "type/JSON"]
                      (when (seq disabled-db-ids)
                        [:not-in :t.db_id disabled-db-ids])]}))
```
</blockquote></details>

I did this to avoid writing the query in three SQL dialects. I'll wait until [this](https://metaboat.slack.com/archives/C04DAKKDMFT/p1680591319535039) conversation resolves to do it
___
*[`src/metabase/driver/mysql.clj` line 54 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7peyiDe6jcWH1Ye5S:-NS9yzrY6HMVhx_23M6K:bget1ri) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/driver/mysql.clj#L54)):*
<details><summary><i>Previously, dpsutton wrote…</i></summary><blockquote>

i see. i'm not sure how i feel about the `json-unfolding-default`. It's far away from here, has a completely separate name that is not discoverable from this one, etc. Props that it's only implemented once though. Tricky
</blockquote></details>

The "turn it off for each json field in your db" feature does seem more useful than a DB-level default. I've asked conor about it [here](https://metaboat.slack.com/archives/C04DAKKDMFT/p1680591319535039). 
___
*[`src/metabase/driver/postgres.clj` line 56 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7qdn58fWtrzVMCJ9x:-NSA9011-Yhz0SHi8foQ:bns2be6) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/driver/postgres.clj#L56)):*
<details><summary><i>Previously, dpsutton wrote…</i></summary><blockquote>

again the same question. Why did the ability to disable on a per-db basis go away?
</blockquote></details>

Same answer as https://reviewable.io/reviews/metabase/metabase/29751#-NS7peyiDe6jcWH1Ye5S
___
*[`src/metabase/models/database.clj` line 359 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7rhBrBY8aCVnQR7kU:-NSA66zY5NTMswIjpvF7:bf2ybmy) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/models/database.clj#L359)):*
<details><summary><i>Previously, dpsutton wrote…</i></summary><blockquote>

i think this is far easier to understand as `(get-in database [:details :json-unfolding] true)` without the fnil.

Also, the `json-unfolding-default` is a bit cumbersome. We have a similar dilemma in persistence: we want to know if (a) the driver supports it, and (b) if this particular instance has it _enabled_.

```clojure
(defmethod driver/database-supports? [:postgres :persist-models]
  [_driver _feat _db]
  true)

(defmethod driver/database-supports? [:postgres :persist-models-enabled]
  [_driver _feat db]
  (-> db :options :persist-models-enabled))
```

I wonder if a similar pattern might help here? You are doing essentially the same thing but using a `json-unfolding-default` function instead of the `database-supports` feature. 
</blockquote></details>

My intention behind using `fnil` was to cover the `{:json-unfolding nil}` case, which had test coverage for mysql here. https://github.com/metabase/metabase/pull/29751/files#diff-48e9fa48b2de544a3d08e0d0eb94ca6887d0792c79976319c54f1949b611ec6bL605-L618

Without migrations on the `:details` JSON blob, I can't guarantee that this doesn't exist in some customers application DB somewhere.

We need `json-unfolding-default` because it's a different concept to `driver/database-supports?`, and it is necessary that they can have different values. The default is now just a default for new fields. JSON unfolding is now set on the field-level, not the database-level. However, that could change with depending on how [this conversation with conor resolves](https://metaboat.slack.com/archives/C04DAKKDMFT/p1680591319535039)
___
*[`test/metabase/api/field_test.clj` line 756 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7uxZhBlUXsq8PA6NN:-NSA86ARAEX0-iW7Fwll:b-r7rauj) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/test/metabase/api/field_test.clj#L756)):*
<details><summary><i>Previously, dpsutton wrote…</i></summary><blockquote>

this is called `enable-json-unfolding` but the argument dictates if it is enabled or not. note the strange convention of `(enable-json-unfolding! false)`. maybe rename to `set-json-unfolding`
</blockquote></details>

Good point, I renamed to `set-json-unfolding!` in 2f4f401

**Comment by calherries** [comment]
@dpsutton @qnkhuat I've updated this feature so that the DB-level setting overrides the field-level setting when it's turned off. Can one of you (or both) take another look?

**Review by calherries — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29751)* status: 0 of 41 files reviewed, 7 unresolved discussions (waiting on @dpsutton and @qnkhuat)

___
*[`src/metabase/models/database.clj` line 359 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7rhBrBY8aCVnQR7kU:-NSLAAgZ39sqX66XEaQl:b-hq9278) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/models/database.clj#L359)):*
<details><summary><i>Previously, calherries (Cal Herries) wrote…</i></summary><blockquote>

My intention behind using `fnil` was to cover the `{:json-unfolding nil}` case, which had test coverage for mysql here. https://github.com/metabase/metabase/pull/29751/files#diff-48e9fa48b2de544a3d08e0d0eb94ca6887d0792c79976319c54f1949b611ec6bL605-L618

Without migrations on the `:details` JSON blob, I can't guarantee that this doesn't exist in some customers application DB somewhere.

We need `json-unfolding-default` because it's a different concept to `driver/database-supports?`, and it is necessary that they can have different values. The default is now just a default for new fields. JSON unfolding is now set on the field-level, not the database-level. However, that could change with depending on how [this conversation with conor resolves](https://metaboat.slack.com/archives/C04DAKKDMFT/p1680591319535039)
</blockquote></details>

I've moved json-unfolding-default here, and now it's referenced from the `driver/database-supports?` implementations. I've also made the implementation easier to understand, without `fnil`.
 https://github.com/metabase/metabase/blob/0904246419bc20e3a57f19647fec30b71ce17ea4/src/metabase/driver/common.clj#L455

**Review by calherries — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/29751)* status: 0 of 41 files reviewed, 7 unresolved discussions (waiting on @dpsutton and @qnkhuat)

___
*[`src/metabase/driver/mysql.clj` line 54 at r8](https://reviewable.io/reviews/metabase/metabase/29751#-NS7peyiDe6jcWH1Ye5S:-NSLAXSPE-zj3nIFsedA:b-oufjfy) ([raw file](https://github.com/metabase/metabase/blob/2b594cbf421abe03d55ecc16f86a0f59cb16746f/src/metabase/driver/mysql.clj#L54)):*
<details><summary><i>Previously, calherries (Cal Herries) wrote…</i></summary><blockquote>

The "turn it off for each json field in your db" feature does seem more useful than a DB-level default. I've asked conor about it [here](https://metaboat.slack.com/archives/C04DAKKDMFT/p1680591319535039). 
</blockquote></details>

`json-unfolding-default` is referenced from here now. It's in `driver.common` because it will need to be used for every implementation of `driver/database-supports?` for `:nested-field-columns`

File: test/metabase/api/field_test.clj
Comment (qnkhuat) [inline test/metabase/api/field_test.clj:767]: nit: for readability I'll put a check that the json unfolding for db is enabled here.

File: test/metabase/api/field_test.clj
Comment (qnkhuat) [inline test/metabase/api/field_test.clj]: 👏 for structuring this test with readable helpers fns.

File: test/metabase/api/field_test.clj
Comment (qnkhuat) [inline test/metabase/api/field_test.clj:774]: should we have a check that all json fields are folded? this test currently only checks the `json-bit` field.

Comment (calherries) [reply test/metabase/api/field_test.clj:774]: I wouldn't consider that a valuable addition. In what world could changing the database-level setting only affect a subset of JSON fields in a database?

File: test/metabase/api/field_test.clj
Comment (calherries) [reply test/metabase/api/field_test.clj:767]: does that aid readability? To me that just adds more code to read. Line 751 clearly sets json-unfolding for the DB.

**Review by qnkhuat — APPROVED** [review]
I reviewed the new changes(compared to the last PR) and it makes sense to me.
Went through the other changes too but I didn't check it thoroughly since I also reviewed the last PR.

File: src/metabase/db/custom_migrations.clj
Comment (dpsutton) [inline src/metabase/db/custom_migrations.clj:9]: change in docstring introduced a typo

File: test/metabase/driver/postgres_test.clj
Comment (dpsutton) [inline test/metabase/driver/postgres_test.clj:497]: lol not your original mistake but `(:take 1 <threaded-value>)` is operating as `identity` here as `1` is not associative and therefore `:take` as a function is returning the not-found threaded value. Let's fix this. (also take 1 and first together seem a bit duplicative in the first place)

**Comment by dpsutton** [comment]
@calherries I have a question about how we set `:json_unfolding` on a field. If a nested field is itself json, do we want to set the `:json_unfolding` bit on that? Consider a json field `f` that has the shape `{:a {:b 1}}`. This will populate three fields:
1. `f` the "real" field with `:json_unfolding` true which has value `{"a": {"b": 1}}`
2. `f → a` the top level field which is itself json which has value `{"b": 1}` and
3. `f → a → b` which has the value `1`

Currently `f` has `json_unfolding` set to true and the nested fields set to false. Should `f → a` have json unfolding also set to true?

**Comment by calherries** [comment]
> Should `f → a` have json unfolding also set to true?

For now, we don't support json unfolding at the nested field level. You can only toggle json unfolding at the top-level. To support nested field json unfolding was milestone 3 of [the epic](https://github.com/metabase/metabase/issues/28667), but it's still debatable whether it's worth implementing now or later, since it was never asked for in the original issue. To implement it, I would probably make [`flattened-row`](https://github.com/metabase/metabase/blob/4ce364e799c0ed4426d0ffa109585fed49cf4f3e/src/metabase/driver/sql_jdbc/sync/describe_table.clj#L258) take an extra argument, a list of paths that shouldn't be unfolded. But I wanted to get this landed before thinking further.

> What's the difference between json-unfolding and json_unfolding?

`json-unfolding` can be either two things: The key in a result from `describe-table`, or the `json-unfolding` key in the `metabase_database.details` JSON blob. `json_unfolding` is the result of writing the key from `describe-table` when it's written to the `metabase_field`. That happens [here](https://github.com/metabase/metabase/blob/4ce364e799c0ed4426d0ffa109585fed49cf4f3e/src/metabase/sync/sync_metadata/fields/sync_instances.clj#L39) in `insert-new-fields!`. So the same problem exists for every key from `describe-table`.

So I think the only thing special about this is that `json-unfolding` is a key in `metabase_database.details` and `json_unfolding` is a column in `metabase_field`. But I think these are the best names for what they describe.

File: src/metabase/driver/sql_jdbc/sync/describe_table.clj
Comment (calherries) [reply src/metabase/driver/sql_jdbc/sync/describe_table.clj]: Updated here to be a little more explicit 
https://github.com/metabase/metabase/pull/29751/commits/8f52cbe88bce8c1a54b9dec0f92f76d0ca966682#diff-e4db3baaafcba181fc735b94c34048deb437fc47e8a2bd3acb9057732d92b330R408-R410

File: resources/migrations/000_migrations.yaml
Comment (calherries) [reply resources/migrations/000_migrations.yaml:14315]: `validCheckSum: ANY` wasn't supposed to be there. I used it in development to edit the SQL. And removing `validCheckSum` doesn't change the checksum, no. I just tested by removing validCheckSum from a migration and starting the server again. No issues.

**Pull request merged**
